### PR TITLE
Allow setting content encoding and a region in Post Policy.

### PR DESCRIPTION
**Pull request by MastaP**
The current Post Policy class is very restrictive. AWS API allows more fields to be included in the conditions. http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html
This PR enables adding Content-Encoding and also a region.
As this class is public, it can be (and is) used without Minio Client in certain applications. Our use case: prepare the policy in one service and send the form data to another service which works both with S3 and Minio endpoints.

### CODE REVIEW

File: api/src/main/java/io/minio/PostPolicy.java [hunk 1]
```diff
@@ -187,7 +217,7 @@ public String bucketName() {
     formData.put("x-amz-algorithm", ALGORITHM);
 
     DateTime date = new DateTime();
-    String region = BucketRegionCache.INSTANCE.region(this.bucketName);
+    String region = this.region != null ? this.region : BucketRegionCache.INSTANCE.region(this.bucketName);
```
Comment (balamurugana) [inline api/src/main/java/io/minio/PostPolicy.java]: As we already got the bucket name and its region, could you tell me why do we need explicit region setting?

Comment (MastaP) [inline api/src/main/java/io/minio/PostPolicy.java]: No, if PostPolicy is used independently (without the Client), BucketRegionCache is inaccessible

Comment (balamurugana) [inline api/src/main/java/io/minio/PostPolicy.java]: :+1:

Comment (MastaP) [inline api/src/main/java/io/minio/PostPolicy.java]: Any chance to have it merged and released soon? :) Thanks!

Comment (balamurugana) [inline api/src/main/java/io/minio/PostPolicy.java]: Introducing region setter would confuse existing users as nowhere in minio-java we ask region.

To make it compatible, have `formData()` method to be overloaded with another version like
```
public Map<String,String> formData(String accessKey, String secretKey, String region)
public Map<String,String> formData(String accessKey, String secretKey)
```

where the second one uses the first one with `BucketRegionCache`.

Hope this works for you.

**Comment by MastaP** [comment]
@balamurugana Done. In order to keep the original formData() intact, had to extract makeFormData() since the region null-check adds 'throws InvalidArgumentException' to the new method signature.

**Comment by MastaP** [comment]
@harshavardhana 
there are two public methods,
1. `public Map<String,String> formData(String accessKey, String secretKey)`
works as before, fills the region from the cache,
2. `public Map<String,String> formData(String accessKey, String secretKey, String region)`
throws `new InvalidArgumentException("empty region")` if no region is provided.

**Pull request merged**